File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
N = int(input())
count = 0

if N % 5 == 0:
    print(N // 5)
else:
    while True:
        N -= 3
        count += 1

        if N % 5 == 0:
            count += N // 5
            print(count)
            break
        elif N == 1 or N == 2:
            print(-1)
            break

File: 백준/Silver/2839. 설탕 배달/설탕 배달.kt
fun main() = with(System.`in`.bufferedReader()) {
    var n = readln().toInt()
    var count = 0

    while (true) {
        if (n - 5 >= 0 && (n % 5 == 3 || n % 5 == 0)) {
            n -= 5
            count++
            continue
        } else if (n - 3 >= 0) {
            n -= 3
            count++
            continue
        } else if (n == 0) {
            print(count)
            return
        } else {
            print(-1)
            return
        }

    }
}